jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -342,12 +342,16 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
         # trace representing the spectrum itself.
         result = super().add_data(data, color, alpha, **layer_state)
 
-        # Index of spectrum trace plotted by the super class. It is
-        # for now, always the last in the array of marks. This will
-        # have to be reworked once multiple spectra can be over-plotted.
-        # The uncert and mask trace pointers are meant to facilitate this
-        # eventual upgrade.
-        self.data_trace_pointer = len(self.figure.marks) - 1
+        # Index of spectrum trace plotted by the super class. It is the
+        # **latest** item in figure.marks that is a Line instance
+        for ind, mark in reversed(list(enumerate(self.figure.marks))):
+            # we'll use __class__.__name__ since other entries (spectral lines,
+            # etc) defined in jdaviz.core.marks inherit from Lines
+            if mark.__class__.__name__ == 'Lines':
+                self.data_trace_pointer = ind
+                break
+        else:
+            raise ValueError("could not find mark for added data")
```

Comment:
just in case, but in reality this really should never happen and is essentially impossible to include test coverage